mesonbuild/mcompile.py

```diff
@@ -32,7 +32,7 @@
     import argparse
 
 def array_arg(value: str) -> T.List[str]:
-    return UserArrayOption(None, value, allow_dups=True, user_input=True).value
+    return T.cast(T.List[str], UserArrayOption(None, value, allow_dups=True, user_input=True).value)
```

Comment:
The cast is required because of `follow_imports = skip` and thus `value` is of type `Any`, which is an error with the current mypy options. I will drop it if `follow_imports = silent` fixes this.